commit 5710ca53404c286fb80b20229422793ea191d924
Author: tyabus <tyabustest@gmail.com>
Date:   Mon Feb 21 00:57:32 2022 +0600

    Dont do watersplash if we're spectator

diff --git a/src/game/server/hl2/hl2_player.cpp b/src/game/server/hl2/hl2_player.cpp
index e5a729b4..0b145ea3 100644
--- a/src/game/server/hl2/hl2_player.cpp
+++ b/src/game/server/hl2/hl2_player.cpp
@@ -3713,6 +3713,9 @@ const impactdamagetable_t &CHL2_Player::GetPhysicsImpactDamageTable()
 //-----------------------------------------------------------------------------
 void CHL2_Player::Splash( void )
 {
+	if( IsObserver() )
+		return;
+
 	CEffectData data;
 	data.m_fFlags = 0;
 	data.m_vOrigin = GetAbsOrigin();
