commit 592e37a832501b85d780c50582ee288433512730
Author: tyabus <tyabustest@gmail.com>
Date:   Tue May 3 20:34:19 2022 +0600

    Forcefully spawn former spectator after team change

diff --git a/src/game/server/hl2mp/hl2mp_client.cpp b/src/game/server/hl2mp/hl2mp_client.cpp
index 02479fa9..0d5721b0 100644
--- a/src/game/server/hl2mp/hl2mp_client.cpp
+++ b/src/game/server/hl2mp/hl2mp_client.cpp
@@ -70,9 +70,9 @@ void FinishClientPutInServer( CHL2MP_Player *pPlayer )
 	pPlayer->ShowViewPortPanel( PANEL_INFO, true, data );
 
 	if ( HL2MPRules()->IsTeamplay() == true )
-        {
+	{
                 ClientPrint( pPlayer, HUD_PRINTTALK, "You are on team %s1\n", pPlayer->GetTeam()->GetName() );
-        }
+	}
 
 	data->deleteThis();
 }
diff --git a/src/game/server/hl2mp/hl2mp_player.cpp b/src/game/server/hl2mp/hl2mp_player.cpp
index 158d84be..4d6e1548 100644
--- a/src/game/server/hl2mp/hl2mp_player.cpp
+++ b/src/game/server/hl2mp/hl2mp_player.cpp
@@ -917,6 +917,7 @@ bool CHL2MP_Player::BumpWeapon( CBaseCombatWeapon *pWeapon )
 void CHL2MP_Player::ChangeTeam( int iTeam )
 {
 	bool bKill = false;
+	bool bWasSpectator = false;
 
 	if ( HL2MPRules()->IsTeamplay() != true && iTeam != TEAM_SPECTATOR )
 	{
@@ -932,6 +933,11 @@ void CHL2MP_Player::ChangeTeam( int iTeam )
 		}
 	}
 
+	if (this->GetTeamNumber() == TEAM_SPECTATOR)
+	{
+		bWasSpectator = true;
+	}
+
 	BaseClass::ChangeTeam( iTeam );
 
 	m_flNextTeamChangeTime = gpGlobals->curtime + TEAM_CHANGE_INTERVAL;
@@ -945,6 +951,12 @@ void CHL2MP_Player::ChangeTeam( int iTeam )
 		SetPlayerModel();
 	}
 
+	if ( bWasSpectator )
+	{
+		Spawn();
+		return; // everything is useless afterwards
+	}
+
 	DetonateTripmines();
 	ClearUseEntity();
 
@@ -981,7 +993,7 @@ bool CHL2MP_Player::HandleCommand_JoinTeam( int team )
 		return true;
 	}
 
-	if( GetNextTeamChangeTime() >= gpGlobals->curtime )
+	if( GetNextTeamChangeTime() > gpGlobals->curtime )
 	{
                 char szReturnString[128];
                 Q_snprintf( szReturnString, sizeof( szReturnString ), "Please wait %d more seconds before trying to switch teams again.\n", (int)(GetNextTeamChangeTime() - gpGlobals->curtime) );
@@ -997,6 +1009,19 @@ bool CHL2MP_Player::HandleCommand_JoinTeam( int team )
 		return false;
 	}
 
+	// Don't do anything if you join your own team
+	if (team == GetTeamNumber())
+	{
+		return false;
+	}
+	
+	// end early
+	if (this->GetTeamNumber() == TEAM_SPECTATOR)
+	{
+		ChangeTeam(team);
+		return true;
+	}
+
 	if ( team == TEAM_SPECTATOR )
 	{
 		// Prevent this is the cvar is set
@@ -1006,7 +1031,7 @@ bool CHL2MP_Player::HandleCommand_JoinTeam( int team )
 			return false;
 		}
 
-		if ( GetTeamNumber() != TEAM_UNASSIGNED && !IsDead() )
+		if (GetTeamNumber() != TEAM_UNASSIGNED && !IsDead())
 		{
 			m_fNextSuicideTime = gpGlobals->curtime;	// allow the suicide to work
 
