commit 5cf73a40bf04833d33c743445f00429acc05be7c
Author: tyabus <tyabustest@gmail.com>
Date:   Sat Apr 9 23:38:54 2022 +0600

    Fix damage filters

diff --git a/src/game/server/filters.cpp b/src/game/server/filters.cpp
index 6179254d..c64d2b21 100644
--- a/src/game/server/filters.cpp
+++ b/src/game/server/filters.cpp
@@ -372,7 +372,8 @@ protected:
 
 	bool PassesDamageFilterImpl(const CTakeDamageInfo &info)
 	{
-	 	return info.GetDamageType() == m_iDamageType;
+		// Tony: these are bitflags. check them as so.
+		return ((info.GetDamageType() & m_iDamageType) == m_iDamageType);
 	}
 
 	int m_iDamageType;
