commit 8b996e6693d1566454a164c6c6c049d5a0fd067b
Author: tyabus <tyabustest@gmail.com>
Date:   Sun Apr 10 00:06:07 2022 +0600

    Add noshadow flag to the SetWeaponVisible function

diff --git a/src/game/shared/basecombatweapon_shared.cpp b/src/game/shared/basecombatweapon_shared.cpp
index f7f34301..9193a1ed 100644
--- a/src/game/shared/basecombatweapon_shared.cpp
+++ b/src/game/shared/basecombatweapon_shared.cpp
@@ -1319,17 +1319,21 @@ void CBaseCombatWeapon::SetWeaponVisible( bool visible )
 
 	if ( visible )
 	{
+		RemoveEffects( EF_NOSHADOW );
 		RemoveEffects( EF_NODRAW );
 		if ( vm )
 		{
+			vm->RemoveEffects( EF_NOSHADOW );
 			vm->RemoveEffects( EF_NODRAW );
 		}
 	}
 	else
 	{
+		AddEffects( EF_NOSHADOW );
 		AddEffects( EF_NODRAW );
 		if ( vm )
 		{
+			vm->AddEffects( EF_NOSHADOW );
 			vm->AddEffects( EF_NODRAW );
 		}
 	}
