commit a88f99d6b78630c14fcd78a27de43abe67503f73
Author: tyabus <tyabustest@gmail.com>
Date:   Mon Feb 21 01:04:40 2022 +0600

    Leave vehicle on team change

diff --git a/src/game/server/hl2mp/hl2mp_player.cpp b/src/game/server/hl2mp/hl2mp_player.cpp
index 9ea32dd6..94ab245b 100644
--- a/src/game/server/hl2mp/hl2mp_player.cpp
+++ b/src/game/server/hl2mp/hl2mp_player.cpp
@@ -954,6 +954,11 @@ void CHL2MP_Player::ChangeTeam( int iTeam )
 			FlashlightTurnOff();
 		}
 
+		if ( IsInAVehicle() )
+		{
+			LeaveVehicle();
+		}
+
 		State_Transition( STATE_OBSERVER_MODE );
 	}
 
