commit c2563fde0d0db743909214290768f876c2716ac9
Author: tyabus <tyabustest@gmail.com>
Date:   Sun Apr 10 23:20:16 2022 +0600

    DetonateTripmines on ChangeTeam

diff --git a/src/game/server/hl2mp/hl2mp_player.cpp b/src/game/server/hl2mp/hl2mp_player.cpp
index 01a0e2dc..158d84be 100644
--- a/src/game/server/hl2mp/hl2mp_player.cpp
+++ b/src/game/server/hl2mp/hl2mp_player.cpp
@@ -945,6 +945,7 @@ void CHL2MP_Player::ChangeTeam( int iTeam )
 		SetPlayerModel();
 	}
 
+	DetonateTripmines();
 	ClearUseEntity();
 
 	if ( iTeam == TEAM_SPECTATOR )
