commit cc526a29683044c6ce9e523c2c750958f1a9d528
Author: tyabus <tyabustest@gmail.com>
Date:   Wed Apr 26 15:17:29 2023 +0600

    shared: Fix smg2/ar2 weapon bob

diff --git a/src/game/shared/baseviewmodel_shared.cpp b/src/game/shared/baseviewmodel_shared.cpp
index 20538e8c..d5783cdd 100644
--- a/src/game/shared/baseviewmodel_shared.cpp
+++ b/src/game/shared/baseviewmodel_shared.cpp
@@ -407,15 +407,12 @@ void CBaseViewModel::CalcViewModelView( CBasePlayer *owner, const Vector& eyePos
 	}
 	// Add model-specific bob even if no weapon associated (for head bob for off hand models)
 	AddViewModelBob( owner, vmorigin, vmangles );
-#if !defined ( CSTRIKE_DLL )
-	// This was causing weapon jitter when rotating in updated CS:S; original Source had this in above InPrediction block  07/14/10
-	// Add lag
-	CalcViewModelLag( vmorigin, vmangles, vmangoriginal );
-#endif
 
 #if defined( CLIENT_DLL )
 	if ( !prediction->InPrediction() )
 	{
+		// Add lag
+		CalcViewModelLag( vmorigin, vmangles, vmangoriginal );
 		// Let the viewmodel shake at about 10% of the amplitude of the player's view
 		vieweffects->ApplyShake( vmorigin, vmangles, 0.1 );	
 	}
diff --git a/src/game/shared/hl2mp/weapon_hl2mpbase_machinegun.h b/src/game/shared/hl2mp/weapon_hl2mpbase_machinegun.h
index c7f9a56b..e1b6cb32 100644
--- a/src/game/shared/hl2mp/weapon_hl2mpbase_machinegun.h
+++ b/src/game/shared/hl2mp/weapon_hl2mpbase_machinegun.h
@@ -5,6 +5,7 @@
 //=============================================================================//
 
 #include "weapon_hl2mpbase.h"
+#include "weapon_hl2mpbasehlmpcombatweapon.h"
 
 #ifndef BASEHLCOMBATWEAPON_H
 #define BASEHLCOMBATWEAPON_H
@@ -19,10 +20,10 @@
 //=========================================================
 // Machine gun base class
 //=========================================================
-class CHL2MPMachineGun : public CWeaponHL2MPBase
+class CHL2MPMachineGun : public CBaseHL2MPCombatWeapon
 {
 public:
-	DECLARE_CLASS( CHL2MPMachineGun, CWeaponHL2MPBase );
+	DECLARE_CLASS( CHL2MPMachineGun, CBaseHL2MPCombatWeapon);
 	DECLARE_DATADESC();
 
 	CHL2MPMachineGun();
