commit d8fdd1ffd68e8baf7524d4a77d9192dde69b9018
Author: tyabus <tyabustest@gmail.com>
Date:   Sat Feb 19 23:39:46 2022 +0600

    viewmodel_fov,default_fov cvars are no longer cheat

diff --git a/src/game/client/hl2/hl2_clientmode.cpp b/src/game/client/hl2/hl2_clientmode.cpp
index 26650ce2..fa449650 100644
--- a/src/game/client/hl2/hl2_clientmode.cpp
+++ b/src/game/client/hl2/hl2_clientmode.cpp
@@ -14,7 +14,7 @@
 #include "tier0/memdbgon.h"
 
 // default FOV for HL2
-ConVar default_fov( "default_fov", "75", FCVAR_CHEAT );
+ConVar default_fov( "default_fov", "75", FCVAR_ARCHIVE, "Sets the field-of-view.", true, 0.1, true, MAX_FOV );
 
 // The current client mode. Always ClientModeNormal in HL.
 IClientMode *g_pClientMode = NULL;
diff --git a/src/game/client/view.cpp b/src/game/client/view.cpp
index af2cf4d5..13ca9cf6 100644
--- a/src/game/client/view.cpp
+++ b/src/game/client/view.cpp
@@ -107,13 +107,9 @@ extern ConVar cl_forwardspeed;
 static ConVar v_centermove( "v_centermove", "0.15");
 static ConVar v_centerspeed( "v_centerspeed","500" );
 
-#ifdef TF_CLIENT_DLL
 // 54 degrees approximates a 35mm camera - we determined that this makes the viewmodels
 // and motions look the most natural.
 ConVar v_viewmodel_fov( "viewmodel_fov", "54", FCVAR_ARCHIVE, "Sets the field-of-view for the viewmodel.", true, 0.1, true, 179.9 );
-#else
-ConVar v_viewmodel_fov( "viewmodel_fov", "54", FCVAR_CHEAT, "Sets the field-of-view for the viewmodel.", true, 0.1, true, 179.9 );
-#endif
 ConVar mat_viewportscale( "mat_viewportscale", "1.0", FCVAR_ARCHIVE, "Scale down the main viewport (to reduce GPU impact on CPU profiling)", true, (1.0f / 640.0f), true, 1.0f );
 ConVar mat_viewportupscale( "mat_viewportupscale", "1", FCVAR_ARCHIVE, "Scale the viewport back up" );
 ConVar cl_leveloverview( "cl_leveloverview", "0", FCVAR_CHEAT );
diff --git a/src/game/shared/shareddefs.h b/src/game/shared/shareddefs.h
index 014f64bd..6119d3b6 100644
--- a/src/game/shared/shareddefs.h
+++ b/src/game/shared/shareddefs.h
@@ -230,7 +230,7 @@ enum CastVote
 
 #define MAX_PLACE_NAME_LENGTH		18
 
-#define MAX_FOV						90
+#define MAX_FOV				100
 
 //===================================================================================================================
 // Team Defines
